submariner-operator/submariner-metrics-redesign.md

```diff
@@ -0,0 +1,146 @@
+# Submariner metrics redesign
+
+Related Issue:
+[Exporting Submariner metrics without needing to open ports](https://github.com/submariner-io/submariner/issues/1944)
+
+## Summary
+
+Submariner Gateway and Globanet pods run with HostNetworking. To access metrics exported by these two we need to open ports
+in the infra firewall. Currently design has a limitation where these ports might be in use by some other service,
+causing a conflict and metrics becoming inaccessible. With the move towards minimizing changes needed in cloud infra for
+submariner, there is a use case to export metrics in a way that doesn't require any of these ports to be opened.
+
+## Proposal
+
+This enhancement proposes a redesign of existing metrics for Submariner Gateway and Globalnet by adding a new metrics proxy
+service. Pods running on the same node can access ports without requiring pods to be opened in the cloud infra. Any queries
+for these metrics will land on the proxy which will always run on the same node.
+
+Note that this proposal doesn't address the issue where metrics port is unavailable on the node. That should be addressed as
```

Comment:
Why not address it as part of this work?

We can have different metrics ports with this, so why not choose a free range [1] and use ports from there?
User's shouldn't be affected (since they're using the service), and we'll be solving both the problems mentioned in the summary.

Once the solution is implemented, the scope of work of changing the ports is much smaller, so it can be part of the work here.

[1] https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers#Registered_ports